packages/stores/src/queries/valset-pref/users-validator-prefs.ts

```diff
@@ -0,0 +1,65 @@
+import { KVStore } from "@keplr-wallet/common";
+import {
+  ChainGetter,
+  ObservableChainQuery,
+  ObservableChainQueryMap,
+} from "@keplr-wallet/stores";
+import { computed, makeObservable } from "mobx";
+
+type QueryUserValidatorPreferencesResponse = {
+  preferences: {
+    val_oper_address: string;
+    weight: string;
+  }[];
+};
+
+export class ObservableQueryUserValidatorPreferences extends ObservableChainQuery<QueryUserValidatorPreferencesResponse> {
```

Comment:
What's the difference between validator preferences and user delegations? In `stake/index` we use `cosmosQueries.queryDelegations.getQueryBech32Address` to get the validators. Not sure if those are the same or not

Reply:
I meant to follow up on this today. I think you guys are supposed to integrate with valsetprefs query in staking project, given that you use the valsetpref staking message to stake. We should open this topic in the frontend channel.

Reply:
Agree. It seems to work correctly as coded now, but if we are missing something we def need outside eyes to look at it